src/api-service/__app__/agent_registration/__init__.py

```diff
@@ -107,6 +107,9 @@ def post(req: func.HttpRequest) -> func.HttpResponse:
         )
     node.save()
 
+    # if any tasks were running during an earlier instance of this node, clear them out
+    node.mark_tasks_stopped_early()
```

Comment:
If the supervisor rebooted for target setup, and registers again, does this cancel its task? IOW, does this break user-requested reboots?

Reply:
Concerns addressed by #150, so this makes sense to merge after that.